add hooks for propagating traces across XHR callbacks.

arielbackenroth opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ X] Tests for the changes have been added (for bug fixes / features)
- [ X] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

This enables instrumenting traces across XHR callbacks by capturing the trace context on the callback functions from the send context. Current behavior is that context is not propagated when the callback is called.

Issue Number: N/A


## What is the new behavior?

If implemented in the TracingService, context can now be propagated to XHR callbacks.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

crisbeto committed:
fix(docs-infra): fix broken adev tests

Fixes some adev tests that are broken on main.

add hooks for propagating traces across XHR callbacks.

remove unnecessary formatting

add hooks for propagating traces across XHR callbacks.

remove unnecessary formatting

add hooks for propagating traces across XHR callbacks.

remove unnecessary formatting

JeanMeche commented on packages/common/http/src/xhr.ts:
```diff
@@ -8,9 +8,12 @@
 import {
-  Injectable,
+  inject,
+  Injectable
```
```suggestion
  Injectable,
```

arielbackenroth committed:
refactor(http): add hooks for propagating traces across XHR callbacks.

Enables propagating a trace across XHR callbacks by providing a hook for
wrapping the callback with a function bound to the send trace context.

arielbackenroth committed:
refactor(http): add hooks for propagating traces across XHR callbacks.

Enables propagating a trace across XHR callbacks by providing a hook for
wrapping the callback with a function bound to the send trace context.

arielbackenroth commented:
screwed this up while rebasing in git. will put together a new pull request
